### PR TITLE
[AC-1486] Feature: SM Billing

### PR DESCRIPTION
**Pull request by vvolkgang**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Long-lived feature branch to support billing for Secrets Manager.

Requires clients PR: https://github.com/bitwarden/clients/pull/5747

## Code changes

* **file.ext:** Description of what was changed and why

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

### CODE REVIEW

File: src/Core/Repositories/Noop/NoopServiceAccountRepository.cs
Comment (Thomas-Avery) [inline src/Core/Repositories/Noop/NoopServiceAccountRepository.cs:1]: Can we move this Noop into the SM owned folder directory? 

`src/Core/SecretsManager/Repositories/Noop/NoopServiceAccountRepository.cs`

**Review by Thomas-Avery — COMMENTED** [review]
Looks good from SM changes, one minor ask around moving the Noop.

File: src/Core/Repositories/Noop/NoopServiceAccountRepository.cs
Comment (eliykat) [reply src/Core/Repositories/Noop/NoopServiceAccountRepository.cs:1]: Good call, this has been done.

**Comment by eliykat** [comment]
Moved back into draft until we've addressed the Checkmarx feedback, which @cyprain-okeke is looking into.

**Comment by eliykat** [comment]
@cyprain-okeke I thought we should still have the `target="_blank"` attribute to open the link in a new tab, but we can solve the Checkmarx feedback another way. See #3124.

**Review by cyprain-okeke — APPROVED** [review]
Thanks for the conflict resolution

**Review by eliykat — APPROVED** [review]
✅ each commit is approved in a separate PR or has otherwise been reviewed & approved now
✅ list of change files does not show any unexpected changes

Adding the Hold label until the [clients PR](https://github.com/bitwarden/clients/pull/5747) has been approved. This can be merged when both are ready.

File: util/Migrator/DbScripts/2023-06-08_00_OrgUserReadOccupiedSmSeatCountByOrgId.sql
Comment (eliykat) [inline util/Migrator/DbScripts/2023-06-08_00_OrgUserReadOccupiedSmSeatCountByOrgId.sql:1]: Date of migrator script needs to be updated

**Review by r-tome — DISMISSED** [review]

**Review by r-tome — APPROVED** [review]

**Review by cyprain-okeke — APPROVED** [review]
Looks good to me

File: src/Core/Constants.cs [hunk 1]
```diff
@@ -36,6 +36,7 @@ public static class FeatureFlagKeys
     public const string DisplayEuEnvironment = "display-eu-environment";
     public const string DisplayLowKdfIterationWarning = "display-kdf-iteration-warning";
     public const string TrustedDeviceEncryption = "trusted-device-encryption";
+    public const string SecretsManagerBilling = "sm-ga-billing";
```
[the remark this answers is not in the document]
Comment (eliykat) [reply src/Core/Constants.cs:39]: as discussed this is required and has been put back in

**Pull request merged**